### PR TITLE
Use dask cuda worker when GPU resource > 0

### PR DESCRIPTION
**Pull request by ljstrnadiii**
# TL;DR
If a gpu has been specified in the dask plugin resource, assume we want to use `dask cuda worker` i.e. that dask-cuda is present on the specified image.

## Type
 - [x] Feature
 - [x] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Currently we build an arg to submit to the container and inject values such as threads and memory from the limits specified in the resources object in the dask plugin object. [Dask cuda](https://github.com/rapidsai/dask-cuda) gives us many built in gpu tools and all that is required is to ensure dask cuda is installed and then to start out workers with `dask-cuda-worker`. So, in order to accomodate, we simply check for gpu limits in the dask plugin resources and overwrite the first arg to `dask-cuda-worker`.

### CODE REVIEW

**Comment by ljstrnadiii** [comment]
@bstadlbauer any thoughts would be greatly appreciated 🙏

**Comment by pingsutw** [comment]
@ljstrnadiii is it ready to be review?

**Review by bstadlbauer — COMMENTED** [review]
I'm generally in favor, but I am somewhat cautious about making this the default. 

How about adding a configuration option to be able to specify the (GPU) worker command? You can seen an example of how to do configuration [in the spark plugin](https://github.com/flyteorg/flyteplugins/blob/master/go/tasks/plugins/k8s/spark/config.go)

File: go/tasks/plugins/k8s/dask/dask.go [hunk 1]
```diff
@@ -187,6 +187,11 @@ func createWorkerSpec(cluster plugins.DaskWorkerGroup, defaults defaults) (*dask
 			memory := limits.Memory().String()
 			workerArgs = append(workerArgs, "--memory-limit", memory)
 		}
+		// If limits includes gpu, assume dask cuda worker cli startup
+		// https://docs.rapids.ai/api/dask-cuda/nightly/quickstart.html#dask-cuda-worker
+		if !limits.Name(flytek8s.ResourceNvidiaGPU, "0").IsZero() {
+			workerArgs[0] = "dask-cuda-worker"
+		}
```
[the remark this answers is not in the document]
Comment (ljstrnadiii) [reply go/tasks/plugins/k8s/dask/dask.go:194]: Good point! I'll make changes to instead add the option to specify a gpu worker command.

[the remark this answers is not in the document]
Comment (ljstrnadiii) [reply go/tasks/plugins/k8s/dask/dask.go:194]: OK, after thinking about this for a little and getting somewhat more familiar with the code base(s), tell me if this makes sense:
- go to flyteidl plugins [dask.proto](https://github.com/flyteorg/flyteidl/blob/6a7b31438821190be535cc6bbcb6f2939e5a056a/protos/flyteidl/plugins/dask.proto#L28) and add a field "command".
- update flytekit plugins [here](https://github.com/flyteorg/flytekit/blob/4ad64d3037cc6227019c489b13e29c3aef21bb74/plugins/flytekit-dask/flytekitplugins/dask/models.py#L54) to include the "command" option.
- lastly, update [dask.go](https://github.com/flyteorg/flyteplugins/blob/63e1e45de43c39d0bae7dfb973f0176a18228ecb/go/tasks/plugins/k8s/dask/dask.go#L153) to get default command if not specified.

question: This seems like three separate prs. Is this order correct? update proto, update models, update dask.go?

@bstadlbauer if this sounds good to you I will try to get this in over the next week.

**Comment by eapolinario** [comment]
Hi, we are moving all Flyte development to a monorepo. In order to help the transition period, we're moving open PRs to monorepo automatically and your PR was moved to https://github.com/flyteorg/flyte/pull/4129. Notice that if there are any conflicts in the resulting PR they most likely happen due to the change in the import path of the flyte components.
